src/mui/input/EmbeddedArrayInput.js

```diff
@@ -0,0 +1,161 @@
+        )
+    }
+}
+
+const ArrayElement = ({ fields, inputs, member, index, elStyle }) => {
+    const removeElement = () => fields.remove(index);
+    const containerStyle = { padding: '0 1em 1em 1em' };
+    return (
+        <div style={elStyle} className="EmbeddedArrayInputItemContainer">
+            <div style={containerStyle}>
+                {
+                    React.Children.map(inputs, input => input && (
+                        <div key={input.props.source} className={`aor-input-${input.props.source}`} style={input.props.style}>
+                            <ArrayElementFormField input={input} prefix={member} />
+                        </div>
+                    ))
+                }
+            </div>
+            <FlatButton
+                primary
+                label="Remove"
+                icon={<ActionDeleteIcon />}
+                onClick={removeElement} />
+        </div>
+    )
+}
+
+import Labeled from './Labeled';
+import { required } from '../form/validate';
+
+const isRequired = (validate) => {
+    if (validate === required) return true;
+    if (Array.isArray(validate)) {
+        return validate.includes(required);
+    }
+    return false;
+};
+
+const ArrayElementFormField = ({ input, prefix, ...rest }) => {
+    if (input.props.addField) {
+        if (input.props.addLabel) {
+            return (
+                <Field
+                    {...rest}
+                    {...input.props}
+                    name={`${prefix}.${input.props.source}`}
+                    component={Labeled}
+                    label={input.props.label}
+                    isRequired={isRequired(input.props.validate)}
+                >
+                    { input }
+                </Field>
+            );
+        }
+        return (
+            <Field
+                {...rest}
+                {...input.props}
+                name={input.props.source}
+                component={input.type}
+                isRequired={isRequired(input.props.validate)}
+            />
+        );
+    }
+    if (input.props.addLabel) {
+        return (
+            <Labeled
+                {...rest}
+                label={input.props.label}
+                source={input.props.source}
+                isRequired={isRequired(input.props.validate)}
+            >
+                {input}
+            </Labeled>
+        );
+    }
+    return (typeof input.type === 'string') ? input : React.cloneElement(input, rest);
+};
+
+EmbeddedArrayInput.propTypes = {
+    addField: PropTypes.bool.isRequired,
+    allowEmpty: PropTypes.bool.isRequired,
+    basePath: PropTypes.string,
+    children: PropTypes.node,
+    label: PropTypes.string,
+    meta: PropTypes.object,
+    onChange: PropTypes.func,
+    input: PropTypes.object,
+    resource: PropTypes.string,
+    source: PropTypes.string,
+    arrayElStyle: PropTypes.object
+};
+
+EmbeddedArrayInput.defaultProps = {
+    addField: false,
+    allowEmpty: false,
```

Comment:
I'm setting that to `false` by default because it already contains its field and there's no need to create a field for it. Doesn't it work like that ?